core/amber/src/main/scala/edu/uci/ics/texera/web/resource/GoogleResource.java

```diff
@@ -42,14 +45,24 @@
     private static Credential serviceAccountCredential;
     private static NetHttpTransport httpTransport;
 
+    private static HttpRequestInitializer createHttpRequestInitializer(final HttpRequestInitializer requestInitializer) {
+        // gives timeout on Google APIs.
+        return httpRequest -> {
+            requestInitializer.initialize(httpRequest);
+            httpRequest.setConnectTimeout(GOOGLE_TIMEOUT_IN_MS);
```

Comment:
this is mainly only used in retry logic. for all possible exceptions including timeout, we are currently not handling them if no retry is used. It will fail and the user will not get the message for now. We can change this in the future PRs, where we send all error messages to user.